### DISCUSSION TITLE
how to perform a terraform state rm?

### DISCUSSION THREAD

**Discussion by mteodori** [dflook/terraform-github-actions#350] [Q&A] [answered]
I need to remove some items from the state with terraform state rm, is there a way I can do it using these actions or should I install the terraform CLI just for this need? I have tried using the TERRAFORM_PRE_RUN variable as this should be done before a destroy but it doesn't work as terraform init hasn't been executed yet

**Comment by dflook** [comment, marked as answer]
Terraform v1.7 introduced the [removed block](https://developer.hashicorp.com/terraform/language/resources/syntax#removing-resources) for removing resources from state, which works with these actions.

If you are using an earlier version of Terraform you would need to use the `state rm` command, and there is currently no way to do that with these actions.

**Reply by mteodori** [reply]
thanks a lot! exactly what I needed!
